ystockquote.py

```diff
@@ -470,32 +473,30 @@ def get_historical_prices(symbol, start_date, end_date):
     Returns a nested dictionary (dict of dicts).
     outer dict keys are dates ('YYYY-MM-DD')
     """
+    epoch = datetime(1970, 1, 1)
+    p1 = datetime.strptime(start_date, '%Y-%m-%d')
+    p2 = datetime.strptime(end_date, '%Y-%m-%d')
     params = urlencode({
-        's': symbol,
-        'a': int(start_date[5:7]) - 1,
-        'b': int(start_date[8:10]),
-        'c': int(start_date[0:4]),
-        'd': int(end_date[5:7]) - 1,
-        'e': int(end_date[8:10]),
-        'f': int(end_date[0:4]),
-        'g': 'd',
-        'ignore': '.csv',
+       'period1': int((p1 - epoch).total_seconds()),
+       'period2': int((p2 - epoch + timedelta(days=1)).total_seconds()),
+       'frequency': '1d',
+       'filter': 'history',
     })
-    url = 'http://real-chart.finance.yahoo.com/table.csv?%s' % params
+    url = 'https://finance.yahoo.com/quote/%s/history?%s' % (symbol, params)
     req = Request(url)
     resp = urlopen(req)
-    content = str(resp.read().decode('utf-8').strip())
-    daily_data = content.splitlines()
+    content = resp.read()
+    quotes = re.findall('{"date":\d+[^}]+}', content)
```

Comment:
Here you would need to call `.decode()` on `content` before passing it to `re.findall`.
It seems that you would also have to check if it's Python 3 and call `.decode()` only if it's Python 3. There may be other, more elegant, way of doing this -- I'm not used to port Python code.